src/pycram/ros/viz_marker_publisher.py

```diff
@@ -107,3 +108,214 @@ def _stop_publishing(self) -> None:
+
+        self.marker_array = MarkerArray()
+        self.marker_overview = {}
+        self.current_id = 0
+
+        self.bw_object = None
+        self.color = None
+        self.pose = None
+        self.name = None
+        self.interval = interval
+
+    def publish(self, pose: Pose, color=None, bw_object=None, name=None):
+        """
+        Publish a pose or an object into the MarkerArray.
+        Priorities to add an object if possible
+
+        :param pose: Pose of the marker
+        :param color: Color of the marker if no object is given
+        :param bw_object: Object to add as a marker
+        :param name: Name of the marker
+        """
+
+        if color is None:
+            color = [1, 0, 1, 1]
+        self.color = color
+        self.name = name
+        self.pose = pose
+        self.bw_object = bw_object
+
+        self.start_time = time.time()
+        thread = threading.Thread(target=self._publish)
+        thread.start()
+        thread.join()
+
+    def _publish(self):
+        """
+        Publish the marker into the MarkerArray
+        """
+        stop_thread = False
+        duration = 2
+
+        while not stop_thread:
+            if time.time() - self.start_time > duration:
+                stop_thread = True
+            if self.bw_object is None:
+                self._publish_pose(name=self.name, pose=self.pose, color=self.color)
+            else:
+                self._publish_object(name=self.name, pose=self.pose, bw_object=self.bw_object)
+
+            rospy.sleep(self.interval)
+
+    def _publish_pose(self, name, pose, color=None):
+        """
+        Publish a Pose as a marker
+
+        :param name: Name of the marker
+        :param pose: Pose of the marker
+        :param color: Color of the marker
+        """
+
+        if name is None:
+            name = 'pose_marker'
+
+        if name in self.marker_overview.keys():
+            self._update_marker(self.marker_overview[name], new_pose=pose)
+            return
+
+        color_rgba = ColorRGBA(*color)
+        self.create_marker(name=name, marker_type=Marker.ARROW, marker_pose=pose,
+                           marker_scales=(0.05, 0.05, 0.05), color_rgba=color_rgba)
+        self.marker_array_pub.publish(self.marker_array)
+        rospy.logwarn("Marker array published")
+
+    def _publish_object(self, name, pose, bw_object):
+        """
+        Publish an Object as a marker
+
+        :param name: Name of the marker
+        :param pose: Pose of the marker
+        :param bw_object: Bulletworld object for the marker
+        """
+
+        bw_real = bw_object.resolve()
+
+        if name is None:
+            name = bw_real.name
+
+        if name in self.marker_overview.keys():
+            self._update_marker(self.marker_overview[name], new_pose=pose)
+            return
+
+        path = bw_real.bullet_world_object.urdf_object.links[0].visual.geometry.filename
```

Comment:
bullet_world_object was changed to world_object in the AbstractWorld merge.

This will also just work for Objects that are spawned from meshes, meaning Objects that have more than one link will not be displayed correctly.